src/assets/js/graphThing.js

```diff
@@ -1,19 +1,160 @@
+var self;
+var med;
+
 export default class Graph {
   constructor (mediator) {
     this.mediator = mediator;
+    med = this.mediator;
+    self = this;
+
+    return this;
   }
 
   init () {
-    console.log('fix the Graph')
+    console.log('fix the Graph');
+
+    med.ee.on("graph:toggle", self.snatch)
+
+    med.ee.on("sfu:peer", function(ev) {
+      console.log("peer added", ev);
+      if(med.graphedVideo) med.ee.emit("graph:toggle", "no data");
+    });
+
+    med.ee.on('speaking', function(data) {
+      console.log('speaking', data);
+      if(med.graphedVideo) {
+        var videoArray = document.querySelectorAll('video');
+
+        videoArray.forEach((item, i) => {
+          //if this video is the one speaking
+          if(item.id == data.socketId + '-video') {
+            // check if it's not already speaking
+            if(item.classList.contains('speaking')){
+              // already speaking
+               return;
+             } else {
+               // turn speaking on (makes it bigger and orange)
+               item.classList.add('speaking');
+             }
+          } else {
+            item.classList.remove('speaking');
+          }
+        });
+        self.snatch();
+      }
+    });
+  }
+
+  async snatch (data) {
+    if(!med.graphedVideo) {
+      var div = document.createElement('div');
+      var selfview = document.getElementById('local');
+      selfview.classList.remove('local-video');
+      selfview.classList.remove('clipped');
+      div.appendChild(selfview);
+      document.getElementById('grid').appendChild(div);
+    }
+    med.graphedVideo = true;
+    self.graphWorker = new Worker('/assets/workers/workerGraph.js');
+    /* Prototype
+    *
+    *
+    *
+
+    // create Graph Div to make the graph area??
```

Comment:
Think we can remove dead code, unless you really going to use this in the future we can keep and uncomment